<a href="https://colab.research.google.com/github/spockthompson/Project-2-Machine-Learning/blob/main/Project_2_Part_1_(Core).ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# First choice: dataset 1
 - https://www.kaggle.com/datasets/deepu1109/star-dataset
 - This data set uses key features to predict what type of star it is.

In [1]:
import pandas as pd
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [2]:
df1 = pd.read_csv('/content/6 class csv.csv')
df1.head()

Temperature (K)  Luminosity(L/Lo)  Radius(R/Ro)  Absolute magnitude(Mv)  \
0             3068          0.002400        0.1700                   16.12   
1             3042          0.000500        0.1542                   16.60   
2             2600          0.000300        0.1020                   18.70   
3             2800          0.000200        0.1600                   16.65   
4             1939          0.000138        0.1030                   20.06   

   Star type Star color Spectral Class  
0          0        Red              M  
1          0        Red              M  
2          0        Red              M  
3          0        Red              M  
4          0        Red              M

In [3]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 240 entries, 0 to 239
Data columns (total 7 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Temperature (K)         240 non-null    int64  
 1   Luminosity(L/Lo)        240 non-null    float64
 2   Radius(R/Ro)            240 non-null    float64
 3   Absolute magnitude(Mv)  240 non-null    float64
 4   Star type               240 non-null    int64  
 5   Star color              240 non-null    object 
 6   Spectral Class          240 non-null    object 
dtypes: float64(3), int64(2), object(2)
memory usage: 13.2+ KB


3. What is the target?
 - Star Type
4. What does one row represent? (A person?  A business?  An event? A product?)
 - Lumosity represents the level of light reflection in space from that specific star from earth.
5. Is this a classification or regression problem?
 - Classification
6. How many features does the data have?
 - One
7. How many rows are in the dataset?
 - 240 Rows
8. What, if any, challenges do you foresee in cleaning, exploring, or modeling with this dataset?
 - I don't see much issue with this set, but we won't know for sure until we try.

#Second choice: dataset 2
 - https://www.kaggle.com/datasets/prathamtripathi/drug-classification
 - Drug Classification based on blood pressure and Cholesterol

In [4]:
df2 = pd.read_csv('/drug200.csv')
df2.head()

Age Sex      BP Cholesterol  Na_to_K   Drug
0   23   F    HIGH        HIGH   25.355  DrugY
1   47   M     LOW        HIGH   13.093  drugC
2   47   M     LOW        HIGH   10.114  drugC
3   28   F  NORMAL        HIGH    7.798  drugX
4   61   F     LOW        HIGH   18.043  DrugY

In [5]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 200 entries, 0 to 199
Data columns (total 6 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Age          200 non-null    int64  
 1   Sex          200 non-null    object 
 2   BP           200 non-null    object 
 3   Cholesterol  200 non-null    object 
 4   Na_to_K      200 non-null    float64
 5   Drug         200 non-null    object 
dtypes: float64(1), int64(1), object(4)
memory usage: 9.5+ KB


3. What is the target?
 - The Target would be the Drug type
4. What does one row represent? (A person?  A business?  An event? A product?)
 - BP represents Blood Pressure
5. Is this a classification or regression problem?
 - Classification
6. How many features does the data have?
 - One feature "Drug type"
7. How many rows are in the dataset?
 - Looks to be 200
8. What, if any, challenges do you foresee in cleaning, exploring, or modeling with this dataset?
 - Maybe cleaning up some of the data, I already see some items capitalized and some spacing issues.